ui/src/message_popup/hint_card.jsx

```diff
@@ -0,0 +1,141 @@
+*/
+export default React.createClass({
+  displayName: 'HintCard',
+
+  getInitialState: function(){
+    return ({
+      hidden: true,
+      exampleIndex: 0,
+      nonExampleIndex: 0,
+    });
+  },
+  
+  propTypes: {
+    examples: React.PropTypes.array.isRequired,
+    nonExamples: React.PropTypes.array.isRequired
+  },
+  
+  onNextExample(){
+    var length = this.props.examples.length;
+    var next = this.state.exampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({exampleIndex: next});
+  },
+  
+  onNextNonExample(){
+    var length = this.props.nonExamples.length;
+    var next = this.state.nonExampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({nonExampleIndex: next});
+  },
+  
+  onHintsToggled(){
+    this.setState({ hidden: !this.state.hidden });
+  },
+
+  render() {
+    const {examples, nonExamples} = this.props;
+    const {hidden, exampleIndex, nonExampleIndex} = this.state;
+    return (
+      <div>
+        {hidden &&
+          (<div>
+            <div style={styles.buttonRow}>
+              <RaisedButton
+                onTouchTap={this.onHintsToggled}
+                style={styles.button}
+                secondary={true}
+                label="Show Examples" />
+              <div style={styles.prompt}>Need a hint?</div>
+            </div>
+          </div>)
+        }
+        {!hidden &&
+          (<div>
+            <div style={styles.examplesBox}>
+              <div style={styles.exampleSection}>
+                <div style={styles.exampleTitle}>Good Example</div>
+                <div style={styles.exampleText}>
+                  {examples[exampleIndex]}
+                </div>
+                <RaisedButton
+                  onTouchTap={this.onNextExample}
+                  style={styles.button}
+                  label="Next example" />
+              </div>
+              <div style={styles.exampleSection}>
+                <div style={styles.exampleTitle}>Bad Example</div>
+                <div style={styles.exampleText}>
+                  {nonExamples[nonExampleIndex]}
+                </div>
+                <RaisedButton
+                  onTouchTap={this.onNextNonExample}
+                  style={styles.button}
+                  label="Next example" />
+              </div>
+            </div>
+            
+            <div style={styles.buttonRow}>
+              <RaisedButton
+                onTouchTap={this.onHintsToggled}
+                style={styles.button}
+                secondary={true}
+                label="Hide Examples" />
+            </div>
+          </div>)
+        }
+      </div>
+    );
+  }
+});
+
+
+const styles = {
+  exampleSection: {
+    display: 'block',
+    marginRight: 10
+  },
+  buttonRow: {
+    margin: 10,
+  },
+  prompt: {
+    fontSize: 16,
+    fontWeight: 'bold',
+    display: 'inline-block',
+  },
+  button: {
+    display: 'inline-block',
+    float: 'right'
```

Comment:
In general, see if you can use [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for layout whenever possible over floats.  Flexbox is newer but well-supported now and is simpler to work with for more complex layouts than float.  For this it would be setting `display: 'flex'` on the `buttonRow` and setting an `alignItems` property on it to make the items align to a particular side.

We can talk this out in person too if that's easier :)